forms-flow-web/src/components/Form/Item/View.js

```diff
@@ -368,11 +368,26 @@ const View = React.memo((props) => {
     } else if (user && !user.role.some(el => el === STAFF_DESIGNER)) {
       /* check formRef before calling function of formio */
       if (formRef.current !== null) {
-        const formSupportedIdentityProviders = getFormSupportedIdentityProviders(
-          formRef.current?.formio, 
-          FORM_SUPPORTED_IDENTITY_PROVIDERS_FIELD_NAME, null);
-        if (Array.isArray(formSupportedIdentityProviders)) {
-          setHasFormAccess(hasUserAccessToForm(formSupportedIdentityProviders, user.username));
+        if (formRef.current?.formio 
+          && formRef.current.formio?._form
+          && formRef.current.formio._form?.supportedidp !== undefined) {
+            if (formRef.current.formio._form?.supportedidp === "") {
+              setHasFormAccess(true);
+            } else {
+              setHasFormAccess(
+                hasUserAccessToForm(
+                  formRef.current.formio._form?.supportedidp.split(","),
+                  user.username
+                )
+              );
+            }
+        } else {
+          const formSupportedIdentityProviders = getFormSupportedIdentityProviders(
```

Comment:
@iman-jamali-fw I have kept old logic intake if not meta data set for "supportedidp". It will check form components for the same. It will keep legacy forms work as per requirements until someone update it in meta data.

Reply:
I suggest removing the old logic to keep the code simple. It should be easy to update the forms with the new IDP metadata.

Reply:
Sure Iman, I have removed an old logic.